### DISCUSSION TITLE
Mengkonversi element dalam kolom

### DISCUSSION THREAD

**Discussion by namiraprita** [teamalgoritma/community#224] [Q&A] [answered]
Bagaimana cara mengganti format menjadi standard formatting dari "619k", "2.6m", "Varies with device" menjadi "619,000", "2,600,000" melalui lopping. Terdapat error seperti ini:

**Error in if (substr(z[i], nchar(z[i]), nchar(z[i])) == "M") { : 
  missing value where TRUE/FALSE needed**

![Screen Shot 2022-07-26 at 11 55 08](https://user-images.githubusercontent.com/107253301/180926408-546d44f2-cd0c-4ff7-9890-3e6bd48c30aa.png)

![Screen Shot 2022-07-26 at 11 55 23](https://user-images.githubusercontent.com/107253301/180926437-474fc914-9be0-4598-aec3-a6e8a86ba839.png)

Terima kasih mohon sharing ilmunya

**Comment by idoali** [comment, marked as answer]
Selamat pagi bu,

Ibu bisa mencoba kode berikut 
```{}
library(stringr)

the_function <- function(x){
  y <- ifelse(str_sub(x, -1) %in% c("M", "K"), 
              ifelse(str_sub(x, -1) == "M", 
                     as.numeric(str_sub(x, 1, -2)) * 1000000,
                     as.numeric(str_sub(x, 1, -2)) * 1000),
              NA)
  return(y)
}
```

Fungsi terrsebut menggunakan library `stringr` dan kita menggunakan fungsi `str_sub` dari library tersebut untuk mengekstrak huruf terakhir dari teks. Jika huruf terakhir adalah K, maka akan dikalikan dengan seribu dan jika diakhiri huruf M maka akan dikalikan dengan satu juga. 

Semoga jawabannya bisa membantu bu
